openpype/hosts/maya/plugins/publish/extract_xgen.py

```diff
@@ -0,0 +1,158 @@
+import os
+import copy
+import tempfile
+
+from maya import cmds
+import xgenm
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api.lib import (
+    maintained_selection, attribute_values, write_xgen_file
+)
+from openpype.lib import StringTemplate
+
+
+class ExtractXgenCache(publish.Extractor):
+    """Extract Xgen
+
+    Workflow:
+    - Duplicate nodes used for patches.
+    - Export palette and import onto duplicate nodes.
+    - Export/Publish duplicate nodes and palette.
+    - Export duplicate palette to .xgen file and add to publish.
+    - Publish all xgen files as resources.
+    """
+
+    label = "Extract Xgen"
+    hosts = ["maya"]
+    families = ["xgen"]
+    scene_type = "ma"
+
+    def process(self, instance):
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        staging_dir = self.staging_dir(instance)
+        maya_filename = "{}.{}".format(instance.data["name"], self.scene_type)
+        maya_filepath = os.path.join(staging_dir, maya_filename)
+
+        # Collect nodes to export.
+        duplicate_nodes = []
+        for node, connections in instance.data["xgenConnections"].items():
+            transform_name = connections["transform"].split(".")[0]
+
+            # Duplicate_transform subd patch geometry.
+            duplicate_transform = cmds.duplicate(transform_name)[0]
+
+            # Discard the children.
+            shapes = cmds.listRelatives(duplicate_transform, shapes=True)
+            children = cmds.listRelatives(duplicate_transform, children=True)
+            cmds.delete(set(children) - set(shapes))
+
+            duplicate_transform = cmds.parent(
+                duplicate_transform, world=True
+            )[0]
```

Comment:
That is true, it wont retain the input connections. But I would argue that you would not want to have it animated since you'll be later connecting it to animation caches.
We could validate against this, but think that is overkill for now.